File: eFormAPI/Plugins/TimePlanning.Pn/TimePlanning.Pn/Infrastructure/Models/PayDayTypeRule/PayDayTypeRuleModel.cs
/*
The MIT License (MIT)

Copyright (c) 2007 - 2021 Microting A/S
*/

namespace TimePlanning.Pn.Infrastructure.Models.PayDayTypeRule;

using System.Collections.Generic;
using PayTimeBandRule;

public class PayDayTypeRuleModel
{
    public int? Id { get; set; }
    public int PayRuleSetId { get; set; }
    public string DayType { get; set; } // Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday, Holiday
    public string DefaultPayCode { get; set; }
    public int Priority { get; set; }
    public List<PayTimeBandRuleModel> TimeBandRules { get; set; } = new List<PayTimeBandRuleModel>();
}

File: eform-client/src/app/plugins/modules/time-planning-pn/models/pay-day-type-rules/pay-day-type-rule.model.ts
import { PayTimeBandRuleModel } from '../pay-time-band-rules/pay-time-band-rule.model';

export class PayDayTypeRuleModel {
  id: number;
  payRuleSetId: number;
  dayType: string;
  defaultPayCode: string;
  priority: number;
  timeBandRules: PayTimeBandRuleModel[];
}
